src/clients/c++/library/grpc_client.cc

```diff
@@ -1046,10 +1048,24 @@ InferenceServerGrpcClient::PreRunProcessing(
         }
       }
     }
+    index++;
   }
 
+  // Remove extra InferInputTensor submessages, that are not required for
+  // this request.
+  if (index < infer_request_.inputs().size()) {
+    infer_request_.mutable_inputs()->DeleteSubrange(
+        index - 1, (infer_request_.inputs().size() - index));
```

Comment:
If, for example, if there is one input, aren't you deleting the subrange that starts at 0? Also, I think just calling RemoveLast the appropriate number of times will be more efficient.

Reply:
My bad. 
> RemoveLast the appropriate number of times will be more efficient.
Is it because the ownership is retained by the array?